dcpy/extract/metadata.py

```diff
@@ -28,12 +30,34 @@ class Template(
     library_dataset: LibraryConfig | None = None
 
 
-def read_template(dataset: str, **kwargs) -> Template:
-    file = TEMPLATE_DIR / f"{dataset}.yml"
+def get_jinja_vars(s: str) -> set[str]:
+    env = jinja2.Environment()
+    parsed_content = env.parse(s)
+    return meta.find_undeclared_variables(parsed_content)
+
+
+def read_template(
+    dataset: str, version: str | None = None, template_dir: Path = TEMPLATE_DIR
+) -> Template:
+    file = template_dir / f"{dataset}.yml"
     logger.info(f"Reading template from {file}")
     with open(file, "r") as f:
         template_string = f.read()
-    template_string = jinja2.Template(template_string).render(**kwargs)
+    vars = get_jinja_vars(template_string)
+    if version is None and len(vars) > 0:
```

Comment:
is empty string a valid version? (ie should we just do `if not version and ...)`

Reply:
Addressed [here](https://github.com/NYCPlanning/data-engineering/pull/654/commits/989b49f1f8b43a31116b85bceda5e5457240545d)